src/common.speech/ServiceRecognizerBase.ts

```diff
@@ -570,38 +574,53 @@ export abstract class ServiceRecognizerBase implements IDisposable {
 
                             // we have a new audio chunk to upload.
                             if (this.privRequestSession.isSpeechEnded) {
-                                // If service already recognized audio end then dont send any more audio
+                                // If service already recognized audio end then don't send any more audio
                                 deferred.resolve(true);
                                 return;
                             }
 
-                            const payload = (audioStreamChunk.isEnd) ? null : audioStreamChunk.buffer;
-                            const uploaded: Promise<boolean> = connection.send(
-                                new SpeechConnectionMessage(
-                                    MessageType.Binary, "audio", this.privRequestSession.requestId, null, payload));
+                            let payload: ArrayBuffer;
+                            let sendDelay: number;
+
+                            if (audioStreamChunk.isEnd) {
+                                payload = null;
+                                sendDelay = 0;
+                            } else {
+                                payload = audioStreamChunk.buffer;
+                                this.privRequestSession.onAudioSent(payload.byteLength);
+
+                                if (maxSendUnthrottledBytes >= this.privRequestSession.bytesSent) {
```

Comment:
>maxSendUnthrottledBytes >= this.privRequestSession.bytesSent [](start = 36, length = 60)

in case the current sent operation crosses the unthrottled boundary, the transaction should be split into the part that gets send immediately and the other part that is throttled.
if not, the services may kick the client in case the part that is passed the unthrottled boundary is too large.

Reply:
This defaults the other way, so that if this frame crosses the boundary it's throttled.

---
In reply to: [286706670](https://github.com/microsoft/cognitive-services-speech-sdk-js/pull/68#discussion_r286706670) [](ancestors = 286706670)